core/tauri-build/src/mobile.rs

```diff
@@ -48,13 +52,40 @@ dependencies {"
 
   app_build_gradle.push_str("\n}");
 
+  if let Some(version) = config.version.as_ref() {
+    app_tauri_properties.push(format!("tauri.android.versionName={}", version));
+    if let Some(version_code) = config.bundle.android.version_code.as_ref() {
+      app_tauri_properties.push(format!("tauri.android.versionCode={}", version_code));
+    } else if let Ok(version) = Version::parse(version) {
+      let version_code = version.major * 1000000 + version.minor * 1000 + version.patch;
+      app_tauri_properties.push(format!("tauri.android.versionCode={}", version_code));
+    }
+  }
```

Comment:
Do we need to check the final version code here? Since it might be out of  the valid range. For example, `create-tauri-app` will create with version `0.0.0` hence the VersionCode is 0 which is invalid

Reply:
@pewsheen This is a good catch! I think it can verified after line 60.

Reply:
I think we should fail when building for android if `versionCode` is not specified in `tauri.conf.json` and app version is `0.0.0`

Reply:
Sounds good! I've created a PR fro this: https://github.com/tauri-apps/tauri/pull/9898